src/cljam/io/util/bin.clj

```diff
@@ -59,3 +60,21 @@
         min-offset (get-min-offset index-data ref-idx beg)]
     (->> (util-chunk/optimize-chunks chunks min-offset)
          (map vals))))
+
+(defn leading-bins-at-level
+  ^long [^long pos ^long level ^long min-shift ^long depth]
+  (unsigned-bit-shift-right pos (+ min-shift (* (- depth level) 3))))
+
+(defn reg->bin
```

Comment:
Would you reconfirm the usage of `reg->bin`? 
I doubt there's a confusion of 0-based half-open and 1-based closed ranges.